test/unit/resources/aws_iam_policy_test.rb

```diff
@@ -0,0 +1,191 @@
+    assert_equal(['test-group'], AwsIamPolicy.new('test-policy-1').attached_groups)
+    assert_nil(AwsIamPolicy.new(policy_name: 'non-existant').attached_groups)
+  end
+
+  def test_property_attached_roles
+    assert_equal(['test-role'], AwsIamPolicy.new('test-policy-1').attached_roles)
+    assert_nil(AwsIamPolicy.new(policy_name: 'non-existant').attached_roles)
+  end
+end
+
+
+#=============================================================================#
+#                               Matchers
+#=============================================================================#
+class AwsIamPolicyMatchersTest < Minitest::Test
+
+  def setup
+    AwsIamPolicy::BackendFactory.select(MAIPSB::Basic)
+  end
+
+  def test_matcher_attachable_positive
+    assert AwsIamPolicy.new('test-policy-1').attachable?
+  end
+
+  def test_matcher_attachable_negative
+    refute AwsIamPolicy.new('test-policy-2').attachable?
+  end
+
+  def test_matcher_attached_positive
+    assert AwsIamPolicy.new('test-policy-1').attached?
+  end
+
+  def test_matcher_attached_negative
+    refute AwsIamPolicy.new('test-policy-2').attached?
+  end
+end
+
+#=============================================================================#
+#                               Test Fixtures
+#=============================================================================#
+module MAIPSB
+  class Empty < AwsIamPolicy::Backend
+    def list_policies(query)
+      OpenStruct.new(policies: [])
+    end
+  end
+
+  class Basic < AwsIamPolicy::Backend
+    def list_policies(query)
+      fixtures = [
+        OpenStruct.new({
+          policy_name: 'test-policy-1',
+          arn: 'arn:aws:iam::aws:policy/test-policy-1',
+          default_version_id: 'v1',
+          attachment_count: 3,
+          is_attachable: true,
+        }),
+        OpenStruct.new({
+          policy_name: 'test-policy-2',
+          arn: 'arn:aws:iam::aws:policy/test-policy-2',
+          default_version_id: 'v2',
+          attachment_count: 0,
+          is_attachable: false,
+        }),
+      ]
+      OpenStruct.new({ policies: fixtures })
+    end
+
+    def list_entities_for_policy(query)
+      policy = {}
+      policy['arn:aws:iam::aws:policy/test-policy-1'] =
+      {
+        policy_groups: [
+          OpenStruct.new({
+            group_name: 'test-group',
+            group_id: 'AIDAIJ3FUBXLZ4VXV34LE',
+          }),
+        ],
+        policy_users: [
+          OpenStruct.new({
+            user_name: 'test-user',
+            user_id: 'AIDAIJ3FUBXLZ4VXV34LE',
+          }),
+        ],
+        policy_roles: [
+          OpenStruct.new({
+            role_name: 'test-role',
+            role_id: 'AIDAIJ3FUBXLZ4VXV34LE',
+          }),
+        ],
+      }
+      policy['arn:aws:iam::aws:policy/test-policy-2'] =
+      {
+        policy_groups: [],
+        policy_users: [],
+        policy_roles: [],
+      }
+      OpenStruct.new( policy[query[:policy_arn]] )
+    end
+  end
+end
```

Comment:
@rx294, this is one of the cleanest, most thorough unit test files I have ever seen.  Excellent work!

Reply:
@rx294 way to go!